Synthetics Bugs

hyperiongeo opened the pull request:
Matt

I had some time so I took a look at this.  Turns out the problem was with the wavelet.  At some point in the past, the input into the wavelet calculation was made in seconds rather than milliseconds but the ricker calculation expected ms.  I changed the ricker calculation to accept seconds (which I believe matches the bruges version).

During testing, I noticed a bug on the as_curve function so I fixed that as well with an improvement over the handling of the log output interval.  I also made a change in the make_synthetic where the user can add their chosen sonic and density mneonics rather than having to rename them to DT and RHOB (which remain the defaults).

Hope this meets with your approval.

Corey

hyperiongeo commented:
I've reviewed this in light of the newer updates and the synthetic issues remain.  This should merge into the existing code without any issues I could identify.

frank1010111 commented on welly/synthetic.py:
```diff
@@ -55,14 +55,19 @@ def basis(self):
         side other curve data.
         """
-        params = {'mnemonic': 'SYN'}
+        params = {'start': start or getattr(self, 'depth_start', None),
```
this will behave unexpectedly if start=0

frank1010111 commented on welly/synthetic.py:
```diff
@@ -55,14 +55,19 @@ def basis(self):
+                  }
 
-        return Curve(data, **params)
+        stop = stop or getattr(self, 'depth_stop', None)
```
this will behave unexpectedly if (albeit this is unlikely) stop = 0

frank1010111 reviewed (commented):
this looks great. I've just got two nit-picks about testing None for truthiness and the obligatory question "do you have unit tests?"

hyperiongeo commented:
Frank - I've reviewed this based on your comments and updated the as_curve function to eliminate the Nones.  Added a few conditionals to account for depths that may not be realistic.  Also added a unit test for the as_curve.
